Sources/SwiftKafka/KafkaConsumer.swift

```diff
@@ -0,0 +1,418 @@
+                return
+            }
+            messageresult = .success(message)
+        } catch {
+            messageresult = .failure(error)
+        }
+
+        let yieldresult = self.messagesSource.yield(messageresult)
+        switch yieldresult {
+        case .produceMore:
+            await self.produceMore()
+        case .dropped, .stopProducing:
+            return
+        }
+    }
+
+    /// Request a new message from the Kafka cluster.
+    /// This method blocks for a maximum of `timeout` milliseconds.
+    /// - Parameter timeout: Maximum amount of milliseconds this method waits for a new message.
+    /// - Returns: A ``KafkaConsumerMessage`` or `nil` if there are no new messages.
+    func poll(timeout: Int32 = 100) async throws -> KafkaConsumerMessage? {
+        // TODO: clock API
+        // TODO: ideally: make private
+        // TODO: is this state needed here?
+        switch self.state {
+        case .started:
+            break
+        case .closed:
+            throw KafkaError(description: "Trying to invoke method on consumer that has been closed.")
+        }
+
+        return try await withCheckedThrowingContinuation { continuation in
+            guard let messagePointer = self.client.withKafkaHandlePointer({ handle in
+                rd_kafka_consumer_poll(handle, timeout)
+            }) else {
+                // No error, there might be no more messages
+                continuation.resume(returning: nil)
+                return
+            }
+
+            defer {
+                // Destroy message otherwise poll() will block forever
+                rd_kafka_message_destroy(messagePointer)
+            }
+
+            // Reached the end of the topic+partition queue on the broker
+            if messagePointer.pointee.err == RD_KAFKA_RESP_ERR__PARTITION_EOF {
+                continuation.resume(returning: nil)
+                return
+            }
+
+            do {
+                let message = try KafkaConsumerMessage(messagePointer: messagePointer)
+                continuation.resume(returning: message)
+            } catch {
+                continuation.resume(throwing: error)
+            }
+        }
+    }
+
+    /// Mark `message` in the topic as read and request the next message from the topic.
+    /// This method is only used for manual offset management.
+    /// - Parameter message: Last received message that shall be marked as read.
+    /// - Warning: This method fails if the `enable.auto.commit` configuration property is set to `true`.
+    public func commitSync(_ message: KafkaConsumerMessage) async throws {
+        switch self.state {
+        case .started:
+            break
+        case .closed:
+            throw KafkaError(description: "Trying to invoke method on consumer that has been closed.")
+        }
+        try await self._commitSync(message)
+    }
+
+    // TODO: commit multiple messages at once -> different topics + test
```

Comment:
What do you think about refactoring this function into a function that takes an array of `KafkaConsumerMessage` as a parameter and commits all of them at once?

This would then always commit the highest offset for each topic.